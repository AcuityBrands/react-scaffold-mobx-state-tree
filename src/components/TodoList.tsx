/**
 * TodoList Component
 * 
 * Renders a list of TodoListItem(s)
 */
import * as React from "react";
import { observer, inject } from "mobx-react";
import {IAppStore} from '../stores'
import { Progress } from 'antd';
import TodoListItem from './TodoListItem';

interface ITodoListProps {
  appStore?: IAppStore
}

@inject("appStore")
@observer
export default class TodoList extends React.Component<ITodoListProps, undefined>{
  render() {
    const { todoStore } = this.props.appStore;
    return (
      <React.Fragment>
        <ul className="todo-list">
          {todoStore.todos.map((todo, idx) => <TodoListItem key={todo.id} todo={todo}/>)}
        </ul>
        <Progress percent={todoStore.completedPct} />
        <br/>
        <div>{todoStore.completedCount} of {todoStore.todoCount} completed</div>
      </React.Fragment>
    );
  }
}