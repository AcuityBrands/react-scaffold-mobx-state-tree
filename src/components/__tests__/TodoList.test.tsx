import * as React from "react";
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { TodoList } from '../TodoList';

Enzyme.configure({ adapter: new Adapter() });

const setup = propOverrides => {
  const props = Object.assign({
    todoStore: {
      todos: [
        { "userId": 1, "id": 1, "title": "todo1", "completed": true },
        { "userId": 2, "id": 2, "title": "todo2", "completed": false }
      ],
      completedCount: 1,
      todoCount: 2
    }
  }, propOverrides)
  const wrapper = Enzyme.shallow(<TodoList {...props} />)
  return {
    props,
    wrapper
  }
}

describe('Todo List Component', () => {
  // Minimal component test confirms component rendered
  it("can render", () => {
    const { wrapper } = setup({});
    expect(wrapper.exists()).toBe(true);
  })

  it("matches the snapshot", () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  })

  it("will render a todo list item for each todo", () => {
    const { wrapper } = setup({});
    const items = wrapper.find("ul").children();
    expect(items.length).toBe(2);
  })

  it("will notify the user of number of completed items", () => {
    const { wrapper } = setup({});
    expect(wrapper.find("div").text()).toEqual("1 of 2 completed");
  })
})