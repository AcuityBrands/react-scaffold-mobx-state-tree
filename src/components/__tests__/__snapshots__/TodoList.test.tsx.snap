// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo List Component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todoStore={
      Object {
        "completedCount": 1,
        "todoCount": 2,
        "todos": Array [
          Object {
            "completed": true,
            "id": 1,
            "title": "todo1",
            "userId": 1,
          },
          Object {
            "completed": false,
            "id": 2,
            "title": "todo2",
            "userId": 2,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul
          className="todo-list"
        >
          <TodoListItem
            todo={
              Object {
                "completed": true,
                "id": 1,
                "title": "todo1",
                "userId": 1,
              }
            }
          />
          <TodoListItem
            todo={
              Object {
                "completed": false,
                "id": 2,
                "title": "todo2",
                "userId": 2,
              }
            }
          />
        </ul>,
        <Progress
          percent={0}
          prefixCls="ant-progress"
          showInfo={true}
          size="default"
          trailColor="#f3f3f3"
          type="line"
        />,
        <br />,
        <div>
          1
           of 
          2
           completed
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TodoListItem
              todo={
                Object {
                  "completed": true,
                  "id": 1,
                  "title": "todo1",
                  "userId": 1,
                }
              }
            />,
            <TodoListItem
              todo={
                Object {
                  "completed": false,
                  "id": 2,
                  "title": "todo2",
                  "userId": 2,
                }
              }
            />,
          ],
          "className": "todo-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "todo": Object {
                "completed": true,
                "id": 1,
                "title": "todo1",
                "userId": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "todo": Object {
                "completed": false,
                "id": 2,
                "title": "todo2",
                "userId": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "percent": 0,
          "prefixCls": "ant-progress",
          "showInfo": true,
          "size": "default",
          "trailColor": "#f3f3f3",
          "type": "line",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            " of ",
            2,
            " completed",
          ],
        },
        "ref": null,
        "rendered": Array [
          1,
          " of ",
          2,
          " completed",
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul
            className="todo-list"
          >
            <TodoListItem
              todo={
                Object {
                  "completed": true,
                  "id": 1,
                  "title": "todo1",
                  "userId": 1,
                }
              }
            />
            <TodoListItem
              todo={
                Object {
                  "completed": false,
                  "id": 2,
                  "title": "todo2",
                  "userId": 2,
                }
              }
            />
          </ul>,
          <Progress
            percent={0}
            prefixCls="ant-progress"
            showInfo={true}
            size="default"
            trailColor="#f3f3f3"
            type="line"
          />,
          <br />,
          <div>
            1
             of 
            2
             completed
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TodoListItem
                todo={
                  Object {
                    "completed": true,
                    "id": 1,
                    "title": "todo1",
                    "userId": 1,
                  }
                }
              />,
              <TodoListItem
                todo={
                  Object {
                    "completed": false,
                    "id": 2,
                    "title": "todo2",
                    "userId": 2,
                  }
                }
              />,
            ],
            "className": "todo-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "todo": Object {
                  "completed": true,
                  "id": 1,
                  "title": "todo1",
                  "userId": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "todo": Object {
                  "completed": false,
                  "id": 2,
                  "title": "todo2",
                  "userId": 2,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "percent": 0,
            "prefixCls": "ant-progress",
            "showInfo": true,
            "size": "default",
            "trailColor": "#f3f3f3",
            "type": "line",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              " of ",
              2,
              " completed",
            ],
          },
          "ref": null,
          "rendered": Array [
            1,
            " of ",
            2,
            " completed",
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
